flask_common/client.py

```diff
@@ -43,7 +43,8 @@ def open(self, *args, **kwargs):
         api_key = kwargs.pop('api_key', self.api_key)
         if 'headers' not in kwargs:
             kwargs['headers'] = self.get_headers(api_key)
-        return super(ApiClient, self).open(*args, **kwargs)
+        with current_app.app_context():
+            return super(ApiClient, self).open(*args, **kwargs)
```

Comment:
The problem with `Client.open` is that it then also applies to a session client, where you potentially want to keep your session for tests. Several tests fail if you have it in `Client`.